lib/screens/category_list_view.dart

```diff
@@ -0,0 +1,130 @@
+    _initData();
+  }
+
+  void _initData() async {
+    _categoryData = await CategoryRepo.categoryDao.getAllCategories();
+    _dataTemplate =
+        await ActivityTemplateRepo.activityTemplateDao.getAllTemplates();
+    print("object...category data is......$_categoryData");
+    setState(() {
+      _dataCategory = _categoryData;
+      print(".......::database data is....${_categoryData}");
+    });
+  }
+
+  Widget _buildButton(
+      BuildContext context, String gameName, String displayName) {
+    MediaQueryData media = MediaQuery.of(context);
+    Orientation orientation = MediaQuery.of(context).orientation;
+    final colors = SingleGame.gameColors[gameName];
+    final color = colors != null ? colors[0] : Colors.amber;
+    var size = media.size;
+    return new Container(
+      decoration: new BoxDecoration(
+        borderRadius: const BorderRadius.all(const Radius.circular(16.0)),
+      ),
+      margin: EdgeInsets.all(size.width * .02),
+      child: new InkWell(
+        onTap: () {
+          String gamename = gameName;
+          String gameid = displayName;
+          Navigator.of(context).push(new MaterialPageRoute(
+              builder: (BuildContext context) =>
+                  new SubcategoryList(gamename: gamename, gameid: gameid)));
+        },
+        key: new Key(gameName),
+        child: new Stack(
+          children: <Widget>[
+            new Material(
+                elevation: 8.0,
+                borderRadius:
+                    const BorderRadius.all(const Radius.circular(16.0)),
+                child: new Container(
+                  decoration: new BoxDecoration(
+                    color: color,
+                    borderRadius:
+                        const BorderRadius.all(const Radius.circular(16.0)),
+                  ),
+                )),
+            new Column(
+              children: <Widget>[
+                new Container(
+                    child: new Container(
+                        child: new Text(Loca.of(context).intl(gameName),
+                            textAlign: TextAlign.right,
+                            textDirection: TextDirection.rtl,
+                            style: new TextStyle(
+                                fontSize: size.height * .04,
+                                color: Colors.white),
+                            overflow: TextOverflow.ellipsis))),
+              ],
+            ),
+          ],
+        ),
+      ),
+    );
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    MediaQueryData media = MediaQuery.of(context);
+    print(media);
+
+    return Container(
+        color: const Color(0xffFECE3D),
+        child: new GridView.count(
+          key: new Key('Category_page'),
+          primary: true,
+          crossAxisSpacing: 12.0,
+          mainAxisSpacing: 12.0,
+          crossAxisCount: media.size.height > media.size.width ? 2 : 2,
+          children: new List.generate(_categoryData.length, (i) {
```

Comment:
what will happen here when the query is running in _initData? _categoryData will be null and you will get an exception